### ISSUE TITLE
We need a function to split Dataset into train / test / eval

### ISSUE THREAD

**Issue by shcheklein** [iterative/datachain#603]
It's a required by customers and is needed for a meaningful prepare data-> train workflow to work. They way we have been doing this so far:

```python
from datachain import DataChain, C

def train_test_split(file) -> str:
    import random
    labels = ["train", "test", "val"]
    return random.choices(labels, weights = [0.7, 0.2, 0.1])[0]

# Add signal

dc = (
    DataChain.from_dataset("qco-image-catalog")
    .map(split=train_test_split, params=["file"], output=str)
    .save()
)

# Save train, test and val datasets

ds_train = (dc.filter(C.split == 'train').save('qco-train'))
ds_test = (dc.filter(C.split == 'test').save('qco-test'))
ds_val = (dc.filter(C.split == 'val').save('qco-val'))
```

Which is a decent start but has a few issues:

- verbose for such a regular use case
- run UDF (compute) for such a simple use case
- might be not stable (each time different results, even if seed is provided - can we even provide a seed and get some meaningful results if we run it in a distributed fashion)

We need a function, probably not on the `dc.py` side, but somewhere in the `lib` that takes a DC and returns a split.

- we can start with wrapping up an existing implementation to iterate on the interface first
- we need later to see / make it work on the database level (no UDF since it's expensive)

Later (in parallel), let's review the existing HuggingFace functionality for this. They have splits as built-in feature. Q: do we want to have something like this on the DC side? Something that looks and feels like a first-class citizen (vs just a wrapper that manipulates with the existing concepts)

**Comment by dmpetrov** [comment]
@shcheklein thank you for creating the issue. I'm just adding a few clarifications.

1. Randomization. We have `sys.rand` in each dataset for exact this reason. The split should be lightweight operation and run inside DB.
2. Seed can be done by XOR-ing: `sys.rand ^ seed` if seed is provided or without any XOR if not provided.
3. We need a to create a special module and populate it with helper functions like this. How about just `toolkit` for now: `from datachain.toolkit import train_test_split`

**Comment by dmpetrov** [comment]
For exact API, please take a look at hugging face implementation and sklearn. Note, we need only basic functionality for now: `train_size, test_size, seed` and `shuffle` is optional for now.

**Comment by dreadatour** [comment]
Follow-up issue: https://github.com/iterative/datachain/issues/606
